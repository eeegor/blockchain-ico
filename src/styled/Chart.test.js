import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Chart } from '.';

describe('Chart', () => {
	it('renders chart wrapper', () => {
		const tree = renderer.create(<Chart>Info</Chart>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders chart wrapper with max width', () => {
		const tree = renderer
			.create(<Chart maxWidth="100">Info</Chart>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('max-width', '100px !important');
	});
});
