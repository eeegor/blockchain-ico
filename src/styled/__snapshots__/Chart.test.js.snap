// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders chart wrapper 1`] = `
.c0 {
  padding: 0;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  position: relative;
  z-index: 7000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow-y: hidden;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
  -webkit-transform: translate3d(0,0,0);
}

.c0 .frame {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  className="c0"
>
  Info
</div>
`;

exports[`Chart renders chart wrapper with max width 1`] = `
.c0 {
  padding: 0;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  position: relative;
  z-index: 7000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow-y: hidden;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
  -webkit-transform: translate3d(0,0,0);
  max-width: 100px !important;
}

.c0 .frame {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  className="c0"
>
  Info
</div>
`;
